Refactor background map texture

benbovy opened the pull request:
Closes #9

Download background map data using `contextily` instead of downloading raw sentinel data. This is much more lightweight, provides the same level of details and also offer more options.

Also tweaked ice remove processing and exposed a few texture arguments so that we can change (and have more control) on the background map texture.

1. A screenshot with the default settings (Esri world imagery tiles):

<img width="943" alt="Screenshot 2024-04-23 at 10 27 49" src="https://github.com/OGGM/oggm-3dviz/assets/4160723/58fa95b4-4723-4cfc-9a46-9f11b8df232a">

2. Another screenshot using the Swiss National Map as texture background:

<img width="942" alt="Screenshot 2024-04-23 at 10 26 25" src="https://github.com/OGGM/oggm-3dviz/assets/4160723/2a3942d8-e4f1-418d-a478-e57f32f0beb2">

The following settings were used for the 2nd screenshot:

```python
from oggm_3dviz.tools import Glacier3DViz
import contextily as cx

viz = Glacier3DViz(
    ds_glacier,
    ...
    use_texture=True,
    texture_args={
        'background_source': cx.providers.SwissFederalGeoportal.NationalMapColor,
        'remove_ice': True,
        'zoom_adjust': 2,
    },
)
```

(actually `remove_ice=True` did work well for the national map background too)

pat-schmitt commented on environment.yml:
```diff
@@ -2,25 +2,21 @@ name: oggm-3dviz
   - scikit-image
-  - stackstac
   - trame
   - trame-vtk
```
I additionally needed to add `- trame-vuetify` to get the interactive jupyter widget working

benbovy commented:
Thanks for the review, @pat-schmitt. I addressed your comments in the last commit.

> I needed to add - trame-vuetify to the environment.yml to get the interactive jupyter widget working (not sure why this was working previously, or what changed)

Maybe this has changed from a required to an optional dependency of one of the `trame-` (conda) packages?

pat-schmitt commented:
Thank you again for this great addition @benbovy!
